src/atomate2/common/flows/magnetism.py

```diff
@@ -0,0 +1,171 @@
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Literal, Sequence
+
+from jobflow import Flow, Maker
+
+from atomate2.common.jobs.magnetism import (
+    enumerate_magnetic_orderings,
+    postprocess_orderings,
+    run_ordering_calculations,
+)
+
+if TYPE_CHECKING:
+    from pymatgen.core import Element
+    from pymatgen.core.structure import Structure
+
+__all__ = ["MagneticOrderingsMaker"]
+
+
+@dataclass
+class MagneticOrderingsMaker(Maker, ABC):
+    """Maker to calculate possible collinear magnetic orderings for a material.
+
+    Given an input structure, possible magnetic orderings will be enumerated and ranked
+    based on symmetry up to a maximum number of orderings. Each ordering will be
+    optionally relaxed and a higher quality static calculation performed to obtain a
+    total energy. The lowest energy ordering is the predicted ground-state collinear
+    ordering.
+
+    This workflow showed decent performance using VASP for a wide range of test
+    materials in a benchmark. It was originally implemented in atomate (v1) for VASP as
+    the MagneticOrderingsWF. Please refer to the following paper for more information
+    and cite appropriately:
+
+        Horton, M.K., Montoya, J.H., Liu, M. et al. High-throughput prediction of the
+        ground-state collinear magnetic order of inorganic materials using Density
+        Functional Theory. npj Computational Materials 5, 64 (2019).
+        https://doi.org/10.1038/s41524-019-0199-7
+
+    .. Note::
+        Good performance of this workflow is ultimately dependent on an appropriate
+        choice of Hubbard U, Hund J values and/or the functional. The defaults will work
+        well for many transition metal oxides.
+
+    Parameters
+    ----------
+    name : str
+        Name of the flows produced by this Maker.
+    static_maker : Maker
+        Maker used to perform static calculations for total energy (e.g., with VASP this
+        would be atomate2.vasp.jobs.StaticMaker).
+    relax_maker : Maker | None
+        Maker used to perform relaxations of the enumerated structures (e.g., with VASP
+        this would be atomate2.vasp.jobs.RelaxMaker). If None, relaxations will be
+        skipped (i.e., only static calculations).
+    default_magmoms : dict | None
+        Optional default mapping of magnetic elements to their initial magnetic moments
+        in µB. Generally these are chosen to be high-spin, since they can relax to a
+        low-spin configuration during a DFT electronic configuration. If None, will use
+        the default values provided in pymatgen/analysis/magnetism/default_magmoms.yaml.
+    strategies : tuple[str]
+        Different ordering strategies to use. Choose from ferromagnetic,
+        antiferromagnetic, antiferromagnetic_by_motif, ferrimagnetic_by_motif,
+        ferrimagnetic_by_species, or nonmagnetic. Here, "motif", means to use a
+        different ordering parameter for symmetry inequivalent sites.
```

Comment:
@mkhorton can you shed some light on this? Not very familiar with this terminology as it is primarily handled in pymatgen. From looking at the `MagneticStructureEnumerator` code, it seems that `ferrimagnetic_by_motif` only is added as a strategy when there is more than one type of wyckoff symbol for the magnetic site, otherwise `ferrimagnetic_by_species` would be the only strategy. Not sure about how results differ

Reply:
My own memory is dim at this point, but I think it's like this:

* Ferrimagnetic orderings require defining at least two distinct sub-lattices (I realize I'm using this term imprecisely), where the spin orientation on the second sub-lattice is opposite to that on the first.
  * `ferrimagnetic_by_species` defines the sub-lattices by saying, e.g., one element is the one sub-lattice and the other element is another sub-lattice. This can work well for materials where there are multiple magnetic species.
  * `ferrimagnetic_by_motif` defines the sub-lattices by taking into account the actual site symmetries, for example consider Fe3O4: in this material there is only a single magnetic element, but there are two distinct sites, an octahedral and a tetrahedral, and the ferrimagnetic order is defined with opposing spins in those two sites.

The default, again, is simply based on how exhaustive you want to be, since the more enumerations the more expensive the calculation. I think the default was set to make it more appropriate for high-throughput.

Reply:
Fantastic response! Thank you for clearing that up